pkg/internal/kube/store.go

```diff
@@ -289,17 +306,36 @@ func (s *Store) serviceNameNamespaceForMetadata(om *informer.ObjectMeta) (string
 	} else {
 		name, namespace = s.serviceNameNamespaceForOwner(om)
 	}
-	if s.sourceLabels.ServiceName != "" {
-		if on, ok := om.Labels[s.sourceLabels.ServiceName]; ok {
-			name = on
+	if nameFromMeta := s.valueFromMetadata(om,
```

Comment:
Doing this here will override any service names set by the OTEL environment variables. I think this will introduce a difference in how it will be handled by the SDK, because environment variables passed to the process directly will, "I think", override whatever else is set. I think it might be better if we added this override in here:
```go
func (s *Store) serviceNameNamespaceOwnerID(ownerKey, name, namespace string) (string, string) {
	serviceName := name
	serviceNamespace := namespace
...
}
```
Right after we set them from whatever k8s found, we can override them based on the labels and annotations and then let the envvar code do its thing.

But we should confirm with the SDKs team if my understanding is correct that the env variables take precedence.

Reply:
Good catch! I reordered the preferences and modified some unit and integration tests to make sure that the environment variables take precedence over annotations or labels.